sql/core/src/main/scala/org/apache/spark/sql/DataFrameNaFunctions.scala

```diff
@@ -468,12 +477,21 @@ final class DataFrameNaFunctions private[sql](df: DataFrame) {
       s"Unsupported value type ${v.getClass.getName} ($v).")
   }
 
+  private def toAttributes(cols: Seq[String]): Seq[Attribute] = {
+    cols.map(df.col(_).named.toAttribute)
```

Comment:
we should be more strict:
```
cols.map(resolve).map {
  case a: Attribute => a
  case _ => fail(is not a top-level column)
}
```

Reply:
Changed as suggested. One naive question: do we need to be more restrictive here even if we use the result of `df.col(_)`, which uses the `outputAttributes` of the plan?

Reply:
`df.col` may return a nested field, which is not an attribute.